### PR TITLE
add `TextLeadingDistribution` to webui `TextStyle`

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Part of https://github.com/flutter/flutter/issues/72521. This adds half leading as a configurable parameter to `TextStyle`.

Canvaskit binding changes: https://skia-review.googlesource.com/c/skia/+/398276/6/modules/canvaskit/paragraph_bindings.cpp

new golden files: https://github.com/flutter/goldens/pull/200/files

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by LongCatIsLooong** [comment]
Getting "Can't load Kernel binary: Invalid kernel binary format version." when I run the tests via felt. Tried deleting `src/out` and reruning `gclient sync -D` but to no avail.

File: lib/web_ui/test/canvaskit/canvas_golden_test.dart [hunk 1]
```diff
@@ -357,6 +357,27 @@ void testMain() {
...
+      await testTextStyle(
+        'half leading inherited from paragraph',
+        height: 20,
+        fontSize: 10,
+        paragraphTextHeightBehavior: ui.TextHeightBehavior(
+          leadingDistribution: ui.TextLeadingDistribution.even,
+        ),
+      );
+      await testTextStyle(
+        'half leading',
```
Comment (yjbanov) [inline lib/web_ui/test/canvaskit/canvas_golden_test.dart:371]: The first argument is used as part of the golden screenshot file name, so it should be distinct from previous tests.

File: lib/web_ui/lib/src/engine/canvaskit/text.dart [hunk 2]
```diff
@@ -87,6 +89,17 @@ class CkParagraphStyle implements ui.ParagraphStyle {
       skStrutStyle.heightMultiplier = style._height;
     }
 
+    switch (style._leadingDistribution ?? paragraphHeightBehavior?.leadingDistribution) {
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/text.dart:92]: nit: I'd extract the switch expression into a variable. Typically we use "effective" as the prefix for something that's resolved from multiple sources, e.g. `effectiveLeadingDistribution`.

File: lib/web_ui/test/canvaskit/canvas_golden_test.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply lib/web_ui/test/canvaskit/canvas_golden_test.dart]: Ops forgot to push. Fixed.

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**